lib/auth/auth_with_roles.go

```diff
@@ -5618,7 +5662,38 @@ func (a *ServerWithRoles) DeleteSAMLIdPServiceProvider(ctx context.Context, name
 		return trace.Wrap(err)
 	}
 
-	return a.authServer.DeleteSAMLIdPServiceProvider(ctx, name)
+	// Get the service provider so we can emit its entity ID later.
+	var entityID string
+	code := events.SAMLIdPServiceProviderDeleteFailureCode
+	sp, deleteErr := a.authServer.GetSAMLIdPServiceProvider(ctx, name)
```

Comment:
I find it a little confusing that the error returned from a _get_ operation is named _deleteErr_. Maybe use `err` for this, and `emitErr` for below?

Reply:
Err whoops. Bad copy paste!